apps/services/auth-public-api/src/app/modules/delegations/actorDelegations.controller.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  BadRequestException,
+  Controller,
+  Get,
+  Query,
+  UseGuards,
+} from '@nestjs/common'
+import {
+  ApiBadRequestResponse,
+  ApiForbiddenResponse,
+  ApiInternalServerErrorResponse,
+  ApiOkResponse,
+  ApiOperation,
+  ApiQuery,
+  ApiTags,
+  ApiUnauthorizedResponse,
+} from '@nestjs/swagger'
+
+import {
+  ActorScopes,
+  AuthMiddlewareOptions,
+  CurrentActor,
+  IdsUserGuard,
+  ScopesGuard,
+} from '@island.is/auth-nest-tools'
+import type { User } from '@island.is/auth-nest-tools'
+import { Audit } from '@island.is/nest/audit'
+import {
+  DelegationDirection,
+  DelegationDTO,
+  DelegationsService,
+} from '@island.is/auth-api-lib'
+import { AuthScope } from '@island.is/auth/scopes'
+import { HttpProblemResponse } from '@island.is/nest/problem'
+
+import { environment } from '../../../environments'
+
+const namespace = '@island.is/auth-public-api/actor/delegations'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@ApiTags('actor-delegations')
+@Controller('v1/actor/delegations')
+@Audit({ namespace })
+export class ActorDelegationsController {
+  constructor(private readonly delegationsService: DelegationsService) {}
+
+  @ActorScopes(AuthScope.actorDelegations)
+  @Get()
+  @ApiOkResponse({ type: [DelegationDTO] })
+  @ApiBadRequestResponse({ type: HttpProblemResponse })
+  @ApiForbiddenResponse({ type: HttpProblemResponse })
+  @ApiUnauthorizedResponse({ type: HttpProblemResponse })
+  @ApiInternalServerErrorResponse()
```

Comment:
Do you believe this is mandatory to specify the default behaviours of a REST api?
Wondering since these swagger decorators are getting really noisy, and ApiBadRequestResponse, ApiForbiddenResponse, ApiForbiddenResponse, and ApiInternalServerErrorResponse don't provide any additional documentation to the route, since they're the behaviours of a REST api.
On the other-side, ApiOkResponse provides a good documentation for this endpoint.

Reply:
This provides documentation on the error type the API returns. As we are returning Problem JSON. But I agree that this is really noisy and was thinking about making a custom decorator `@ApiErrorResponses()` fyrir Nest, sem myndi apply-a öllum þessum ásamt Problem týpunni.

Reply:
sounds much better! Like that approach ... or just really extend swagger as a whole ... have one decorator for swagger:
```typescript
@Documentation({
  response: {
    ok: [DelegationDTO],
    errors: [] // this is provided by default
  },
  description: `Finds all incoming delegations for the signed in user or actor.
			Including the custom delegations as well as natural delegations from NationalRegistry and CompanyRegistry.`,
  query: {
    name: 'direction',
    required: true,
    schema: {
      enum: ['incoming'],
      default: 'incoming',
    },
  },
})
async findAll() {}
```